xml/tuning_cgroups.xml

```diff
@@ -221,8 +220,177 @@
+     Docs: man:mysqld(8)
+           https://mariadb.com/kb/en/library/systemd/
+ Main PID: 11845 (mysqld)
+   Status: "Taking your SQL requests now..."
+    Tasks: 30 (limit: 256)
+   CGroup: /system.slice/mariadb.service
+           └─11845 /usr/sbin/mysqld --defaults-file=/etc/my.cnf --user=mysql
+   </screen>
+   <para>
+       The Tasks line shows that MariaDB currently has 30 tasks running, and has
+       an upper limit of 256. The following example demonstrates how to raise
+       MariaDB's limit to 8096. Create a new override file,
+       <filename>/etc/systemd/system.conf.d/30-mariadb.conf</filename>, 
+       with the following lines, then reload:
+   </para>
+   <screen>
+[Service]
+TasksMax=8096
+   </screen>
+   <screen>&prompt.root;systemctl daemon-reload</screen>
+   <para>
+       The value does not have to be 8096, but can be whatever limit seems
+       appropriate for your workloads, including <literal>infinity</literal>.
+   </para>
+   <para>
+       You may also set custom values for user processes. The &sle; default
+       is <literal>infinity</literal>, set in
+       <filename>/usr/lib/systemd/system/user-.slice.d/20-suse-defaults.conf</filename>.
+       Disable this default by linking it to <filename>/dev/null</filename>
+       in <filename>/etc/systemd/user-.slice.d/</filename>:
+   </para>
+   <screen>&prompt.root;mkdir /etc/systemd/user-.slice.d/
+&prompt.root;ln -s /dev/null /etc/systemd/user-.slice.d/20-suse-defaults.conf     
+   </screen>
+   <para>
+       Create a new override file, 
+       <filename>/etc/systemd/system/user-.slice.d/30-users-tasksmax.conf</filename>,
+       with these lines:
+   </para>
+   <screen>
+[Slice]
+TasksMax=8096
+   </screen>
+   <para>
+       If there is a <literal>UserTasksMax</literal> setting in
+       <filename>/etc/systemd/logind.conf</filename>, comment it out.
+       Then run <command>systemctl daemon-reload</command> to load the new value.
+   </para>
+   <para>
+       To set a custom value for an individual user create an override file, for
+       example 
+       <filename>/etc/systemd/system/user-1000.slice.d/50-user-tasksmax.conf</filename>,
+       with these lines:
+   </para>
+   <screen>
+[Slice]
+TasksMax=8096
+   </screen>
+   <para>
+       Verify the new value:
+   </para>
+   <screen>&prompt.user;systemctl show --property TasksMax user-1000.slice
+TasksMax=8096
+   </screen>
+   <para>
+       The new value takes effect at the next user login.
+   </para>
+   <para>
+       How do you know what values to use? This varies according to your workloads,
+       system resources, and other resource configurations. When your user TasksMax 
+       value is too low, you will see error messages like "Failed to fork (Resources
+       temporarily unavailable)". Test this by setting the user TasksMax value
+       to 128, then open some applications. To recover, close all apps 
+       (except your terminal) and change the setting.
+   </para>
```

Comment:
The ultimate hint for troubleshooting is checking dmesg/syslog for messages from kernel that tells which unit (cgroup) hit the limit:

> [7835271.819498] cgroup: fork rejected by pids controller in /system.slice/run-r61c17f6d334540b599c366514bc1d492.service